src/parse.c

```diff
@@ -0,0 +1,78 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parse.c                                            :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/11/12 16:02:54 by gwolf             #+#    #+#             */
+/*   Updated: 2023/11/20 13:03:50 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "parse.h"
+
+t_err	ft_parse_file(char *filename, t_entities *ents, t_cam *cam)
+{
+	char	**lines;
+
+	lines = NULL;
+	if (ft_import_file(filename, &lines))
+		return (ERROR);
+	if (ft_check_lines(lines, &ents->lsrc_count, &ents->total))
+	{
+		ft_free_char_arr(lines);
+		return (ERROR);
+	}
+	if (ft_malloc_ents(&ents->lsrc, &ents->obj, ents->lsrc_count, ents->total))
+	{
+		ft_free_char_arr(lines);
+		return (ERROR);
+	}
+	ft_parse_lines(ents, cam, lines);
+	ft_free_char_arr(lines);
+	return (SUCCESS);
+}
+
+t_err	ft_malloc_ents(t_light **lsrc, t_hittable **obj, int lsrc_c, int total)
+{
+	if (ft_err_malloc((void **)lsrc, sizeof(**lsrc) * lsrc_c))
+		return (ERROR);
+	if (ft_err_malloc((void **)obj, sizeof(**obj) * total))
+	{
+		free(*lsrc);
+		return (ERROR);
+	}
+	return (SUCCESS);
+}
+
+void	ft_parse_lines(t_entities *ents, t_cam *cam, char **lines)
```

Comment:
I guess this one needs to be shortened at the end

Reply:
True. Was a last minute change to add comments (#), which i saw in your testfile :D
Problem was that comments would inc the id (since it would fall into else clause). Changed to exclude comments.

Reply:
So we don't accept comments in .rt files or how do we handle them?

Reply:
We do: if line starts with # it gets ignored. I just moved some part around to make it fit in 25l